src/components/application_manager/src/commands/mobile/add_sub_menu_request.cc

```diff
@@ -110,14 +110,13 @@ void AddSubMenuRequest::Run() {
                                  [strings::value].asString(),
                       app,
                       application_manager_);
-
     msg_params[strings::sub_menu_icon] =
         (*message_)[strings::msg_params][strings::sub_menu_icon];
-    SendHMIRequest(hmi_apis::FunctionID::UI_AddSubMenu, &msg_params, true);
-  } else {
-    LOG4CXX_ERROR(logger_, "Sub-menu icon is not valid.");
-    SendResponse(false, mobile_apis::Result::INVALID_DATA);
   }
+  if(!is_key_icon_exist) {
+     msg_params.erase(strings::sub_menu_icon);
```

Comment:
@pold500 is is redundant to erase strings::sub_menu_icon, because it is not exists